pkg/cloudprovider/providers/azure/azure_wrap.go

```diff
@@ -40,18 +41,18 @@ var (
 // checkExistsFromError inspects an error and returns a true if err is nil,
 // false if error is an autorest.Error with StatusCode=404 and will return the
 // error back if error is another status code or another type of error.
-func checkResourceExistsFromError(err error) (bool, error) {
+func checkResourceExistsFromError(err error) (bool, string, error) {
 	if err == nil {
-		return true, nil
+		return true, "", nil
 	}
 	v, ok := err.(autorest.DetailedError)
 	if !ok {
-		return false, err
+		return false, "", err
 	}
 	if v.StatusCode == http.StatusNotFound {
-		return false, nil
+		return false, err.Error(), nil
```

Comment:
btw, there is a Message filed in autorest.DetailedError, but its contents are always same. So the error itself is used here